packages/venia-concept/src/components/PurchaseHistory/PurchaseHistoryItem/purchaseHistoryItem.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react';
+import { number, shape, string } from 'prop-types';
+import { Link } from 'react-router-dom';
+
+import Icon from 'src/components/Icon';
+import classify from 'src/classify';
+import defaultClasses from './purchaseHistoryItem.css';
+
+class PurchaseHistoryItem extends Component {
+    static propTypes = {
+        classes: shape({
+            body: string
+        }),
+        item: shape({
+            id: number,
+            imageSrc: string,
+            title: string,
+            date: string,
+            link: string
+        })
+    };
+
+    render() {
+        const {
+            classes,
+            item: { imageSrc, title, date, link }
+        } = this.props;
+
+        const trimedTitle =
+            title.length >= 19 ? `${title.substring(0, 19)}...` : title;
```

Comment:
First of all, why it is `19`? It better to avoid magical numbers. Secondly, it is better to create common util function `trimString(str, maxLength)` or something like this and move the logic there. Thirdly, you can add ellipsis using css `text-overflow: ellipsis;`